github/repos.go

```diff
@@ -767,6 +767,46 @@ type Protection struct {
 	RequiredConversationResolution *RequiredConversationResolution `json:"required_conversation_resolution"`
 }
 
+type ProtectionRules struct {
```

Comment:
Please add godoc comment for this struct and make it singular to match the `"rule"` usage.

```suggestion
type ProtectionRule struct {
```